libnm/nm-device-wireguard.c

```diff
@@ -0,0 +1,254 @@
+
+	_nm_object_register_properties (object,
+	                                NM_DBUS_INTERFACE_DEVICE_WIREGUARD,
+	                                property_info);
+}
+
+static void
+finalize (GObject *object)
+{
+	G_OBJECT_CLASS (nm_device_wireguard_parent_class)->finalize (object);
+}
+
+static void
+get_property (GObject *object,
+              guint prop_id,
+              GValue *value,
+              GParamSpec *pspec)
+{
+	NMDeviceWireguard *device = NM_DEVICE_WIREGUARD (object);
+	NMDeviceWireguardPrivate *priv = NM_DEVICE_WIREGUARD_GET_PRIVATE (object);
+	gchar *b64;
+
+	switch (prop_id) {
+	case PROP_PRIVATE_KEY:
+		b64 = g_base64_encode (priv->private_key, sizeof (priv->private_key));
+		g_value_set_string (value, b64);
+		g_free (b64);
+		break;
+	case PROP_PUBLIC_KEY:
+		b64 = g_base64_encode (priv->public_key, sizeof (priv->public_key));
+		g_value_set_string (value, b64);
+		g_free (b64);
+		break;
+	case PROP_LISTEN_PORT:
+		g_value_set_uint (value, nm_device_wireguard_get_listen_port (device));
+		break;
+	case PROP_FWMARK:
+		g_value_set_uint (value, nm_device_wireguard_get_fwmark (device));
+		break;
+	default:
+		G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+		break;
+	}
+}
+
+static void
+nm_device_wireguard_class_init (NMDeviceWireguardClass *wireguard_class)
+{
+	GObjectClass *object_class = G_OBJECT_CLASS (wireguard_class);
+	NMObjectClass *nm_object_class = NM_OBJECT_CLASS (wireguard_class);
+
+	g_type_class_add_private (wireguard_class, sizeof (NMDeviceWireguardPrivate));
+
+	object_class->finalize = finalize;
+	object_class->get_property = get_property;
+
+	nm_object_class->init_dbus = init_dbus;
+
+	/**
+	 * NMDeviceWireguard:private_key:
+	 *
+	 * 32-byte private key.
+	 * Set to all zeros to purge an existing key.
+	 *
+	 * Since: 1.14
+	 **/
+	obj_properties[PROP_PRIVATE_KEY] =
+	    g_param_spec_value_array (NM_DEVICE_WIREGUARD_PRIVATE_KEY,
+	                              _("Private key"),
+	                              _("32-byte private key. Set to all zeros to purge an existing key."),
+	                              NULL,
+	                              G_PARAM_READABLE | G_PARAM_STATIC_STRINGS);
+
+	/**
+	 * NMDeviceWireguard:public_key:
+	 *
+	 * 32-byte public key, derived from the current private key.
+	 *
+	 * Since: 1.14
+	 **/
+	obj_properties[PROP_PUBLIC_KEY] =
+	    g_param_spec_value_array (NM_DEVICE_WIREGUARD_PUBLIC_KEY,
+	                              _("Public key"),
+	                              _("32-byte public key, derived from the current private key."),
+	                              NULL,
+	                              G_PARAM_READABLE | G_PARAM_STATIC_STRINGS);
+
+	/**
+	 * NMDeviceWireguard:listen-port:
+	 *
+	 * Local UDP listen port.
+	 * Set to 0 to allow a random port to be chosen (default).
+	 *
+	 * Since: 1.14
+	 **/
+	obj_properties[PROP_LISTEN_PORT] =
+	    g_param_spec_uint (NM_DEVICE_WIREGUARD_LISTEN_PORT,
+	                       _("Listen port"),
```

Comment:
In general, we leave these blank. We don't use them anywhere really and a translatable string just adds work for translators.

Reply:
Removing these then.